src/main/java/Deadline.java

```diff
@@ -0,0 +1,59 @@
+import java.time.LocalDateTime;
+import java.time.DateTimeException;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDateTime byDeadline;
+
+    /**
+     * Creates a deadline event.
+     * @param description task description
+     * @param byDeadline deadline for task
+     * @throws DukeException when there is a problem parsing the date time string provided by user
+     */
+    public Deadline(String description, String byDeadline) throws DukeException {
+        super(description);
+        this.taskType = TaskType.DEADLINE;
+        try {
+            this.byDeadline = LocalDateTime.parse(byDeadline, DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm"));

```

Comment:
Would it be a better idea to save the pattern as a final static constant (preferably in Parser class so that you can call it by Parser.DATE_PATTERN)?